### ISSUE TITLE
decorelate platform configuration from data results

### ISSUE THREAD

**Issue by jfhugard** [osadl/Latencyplot#1]
hello,
I create this issue because I think it will be easier to communicate directly via github than by mail for Schneider Electric feedback.

as we try to automate things on a given board it seems interesting to have a JSON file to describe platform 

```
    "system": {
        "hostname": "rack0slot8.osadl.org"
    },
    "processor": {
        "family": "x86",
        "vendor": "Intel",
        "type": "Core i9-9900K"
    },
    "kernel": {
        "version": "5.15.49-rt47",
        "config": [
            "CONFIG_CC_VERSION_TEXT=\"gcc (GCC) 11.3.1 20220421 (Red Hat 11.3.1-2)\"",
            "CONFIG_CC_IS_GCC=y",
            "CONFIG_GCC_VERSION=110301",
            "CONFIG_CLANG_VERSION=0",
.....
```

as this file will remain the same on a given target

and have other JSON files for data results or maybe construct the "final" JSON file by aggregating JSON conf file with data from test.

for the **plot-rtdataset.py** script we can then pass the 2 JSON files to render graphs or no modification if we stay with one "final" JSON file.

Maybe having just one "final" JSON file will help to collect datas from different entities and be sure to have access to all the information within one file.

for the **mk-rtdataset.py** it may imply to separate into 2 differents scripts : 
* one to help generate JSON conf on targets that support python
* one to create JSON "data" file

If you find the idea interesting we should be able to contribute to write / adapt the scripts to implement it.
Plz tell us how you prefer to advance on the topic.

**Comment by osadl** [comment]
Good point. The plot program now has the option to provide a second JSON file that is merged into the first. I took the opportunity of this upgrade change to add parsing of command line arguments and add verbose mode to the program. Also, a list of supported graphics formats can now be created. Finally, the x-scale can be reduced.

The merging operation was a bit difficult. I did a number of tests, but I'm not sure if it always works. If you have time, please help test this feature - preferably with verbosity mode enabled to confirm that unavailable keys are added and available keys are overwritten.

I have not made any changes to the **mk-rtdataset.py** program, nor have I provided an alternate version, because I don't know how you would like to generate the data. You may want to pipe the output of cyclictest directly into the new JSON generator. Could you provide the new JSON generator that meets your requirements?

**Comment by jfhugard** [comment]
Thx for the feedback. We try the last plot script version with our scripts that generate separately `configuration` and `data` and it works. I tried to create a branch with proposed modification but seems  I haven't the right to directly do it... Can you modify the rights for this repo or should I fork it or maybe send you directly our proposed evolution ?

**Comment by osadl** [comment]
Please fork the repo and send a pull request. This is the usual way to collaborate on Github.

**Comment by jfhugard** [comment]
thx for the guidance. pull request created : https://github.com/osadl/Latencyplot/pull/2
Not sure if it ticks all your requirements...  Waiting for you feedback.
